File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/CreateAccountScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Presentation.Scenario;

import java.io.IOException;

public class CreateAccountScenario implements Scenario {
    private final UserService userService;

    public CreateAccountScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Create Account";
    }

    @Override
    public void run() {
        try {
            long accountId = userService.createAccount();

            System.out.println("Successfully created account, id: " + accountId);

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/CreateAccount/CreateAccountScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Contracts.Users.OperationsResults;
using Spectre.Console;

namespace Lab5.Presentation.Console.Scenarios.CreateAccount;

public class CreateAccountScenario : IScenario
{
    private readonly IUserService _userService;

    public CreateAccountScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Create account";

    public void Run()
    {
        CreateAccountResult result = _userService.CreateAccount();

        string message = result switch
        {
            CreateAccountResult.Success success => $"Your account Id is: {success.AccountId.ToString()}",
            CreateAccountResult.NotLoginYet => "Please login to continue",
            _ => throw new ArgumentOutOfRangeException(nameof(result)),
        };

        AnsiConsole.WriteLine(message);
        AnsiConsole.Ask<string>("Ok");
    }
}